config/hypr/hypridle.conf

```
@@ -4,13 +4,21 @@
                                                       
 
 general {
-    # lock_cmd = notify-send "lock!"          # dbus/sysd lock command (loginctl lock-session) 
-    # unlock_cmd = notify-send "unlock!"      # same as above, but unlock
-    before_sleep_cmd = hyprlock    # command ran before sleep
-    # after_sleep_cmd = notify-send "Awake!"  # command ran after sleep
+    lock_cmd = pidof hyprlock || hyprlock # runs hyprlock if it is not already running (this is always run when "loginctl lock-session" is called) 
```

```suggestion
    lock_cmd = pidof hyprlock || hyprlock --immediate # runs hyprlock if it is not already running (this is always run when "loginctl lock-session" is called) 
```

Else there is a bit delay when locking.

---

I just kept it like it was in the listener beforehand.

I thought the flag was just to skip the grace period defined in the `hyprlock.conf`. 

I might be slightly biased as it is a needed feature for me since I personally disable the notifications before locking and have my grace period set to 10 seconds so I can unlock it again fast if I am just sitting there doing nothing.

---

No issues, for me the flag helps mitigate the issue where if I quickly move my mouse right after executing `hyprlock`, the session does not lock properly and quickly unlocks itself.

I am using hyprlock or hypridle configs provided by these dots without any modification.

This very well could be a upstream bug though.

---

> No issues, for me the flag helps mitigate the issue where if I quickly move my mouse right after executing `hyprlock`, the session does not lock properly and quickly unlocks itself.
> 
> I am using hyprlock or hypridle configs provided by these dots without any modification.
> 
> This very well could be a upstream bug though.
> 
> EDIT: this is intended as can be seen from the logs
> 
> ```
> [LOG] In grace and cursor moved more than 5px, unlocking!
> ```

but the unlock immediately was because of grace period aint it?

---

Yes, both for grace period and cursor moving.